peps/pep-0748.rst

```diff
@@ -0,0 +1,923 @@
+sequence of signing chains, which is useful when it is responsible for multiple
+domains.
+
+Trust Store
+~~~~~~~~~~~
+
+As discussed above, loading a trust store represents an issue because different
+TLS implementations vary wildly in how they allow users to select trust stores.
+For this reason, we need to provide a model that assumes very little about the
+form that trust stores take.
+
+This problem is the same as the one that the ``Certificate`` and ``PrivateKey``
+types need to solve. For this reason, we use the exact same model, by creating a
+concrete class that captures the various means of how users could define a trust
+store.
+
+A given TLS implementation is not required to handle all possible trust stores.
+However, it is strongly recommended that a given TLS implementation handles the
+``system`` constructor if at all possible, as this is the most common validation
+trust store that is used. Backends can communicate unsupported options as
+described in the “Runtime” section below.
+
+.. code-block:: python
+
+    TODO TrustStore from tlslib
+
+Runtime Access
+~~~~~~~~~~~~~~
+
+A not-uncommon use case for library users is to want to allow the library to
+control the TLS configuration, but to want to select what backend is in use. For
+example, users of Requests may want to be able to select between OpenSSL or a
+platform-native solution on Windows and macOS, or between OpenSSL and NSS on
+some Linux platforms. These users, however, may not care about exactly how their
+TLS configuration is done.
+
+This poses two problems: given an arbitrary concrete implementation, how can a
+library:
+
+* Work out whether the backend supports particular constructors for certificates
+  or trust stores (e.g. from arbitrary identifiers)?
+
+* Get the correct types for the two context classes?
+
+Constructing certificate and trust store objects should be possible outside of
+the backend. Therefore, the backends need to provide a way for users to verify
+whether the backend is compatible with user-constructed certificates and trust
+stores. Therefore, each backend should implement a ``validate_config`` method
+that takes a ``TLSClientConfiguration`` or ``TLSServerConfiguration`` object and
+raises an exception if unsupported constructors were used.
+
+For the types, there are two options: either all concrete implementations can be
+required to fit into a specific naming scheme, or we can provide an API that
+makes it possible to grab these objects.
+
+This PEP proposes that we use the second approach. This grants the greatest
+freedom to concrete implementations to structure their code as they see fit,
+requiring only that they provide a single object that has the appropriate
+properties in place. Users can then pass this “backend” object to libraries that
+support it, and those libraries can take care of configuring and using the
+concrete implementation.
+
+All concrete implementations must provide a method of obtaining a ``Backend``
```

Comment:
Perhaps TLSImplementation rather than Backend?

At the very least, the TLS prefix should be used here.

Reply:
`TLSImplementation` makes sense to me!